url7895f5cfa7_edm92_defaultlogic_tgz-pJ8-be_fnord_DefaultLogic_AppJ8/src/be/fnord/DefaultLogic/DefaultLogicExample.java

```diff
@@ -0,0 +1,702 @@
+					+ c + "))");
+			a.e.println("= " + world_and_ext.getClosure());
+			a.e.decIndent();
+
+		}
+	}
+
+	/**
+	 * Output Applyig rules to the world Effect:eeee & eeee ==== eeee Given the
+	 * world:
+	 * <p/>
+	 * And the rules [([]):(C) ==> (~D)] , [([]):(D) ==> (~E)] , [([]):(E) ==>
+	 * (~F)] Possible Extensions Ext: Th(W U (~D & ~F))
+	 */
+	public static void example7() {
+		WorldSet myWorld = new WorldSet(); // Empty World
+		myWorld.addFormula(a.e.EMPTY_FORMULA);
+
+		DefaultRule rule1 = new DefaultRule();
+		rule1.setPrerequisite(a.e.EMPTY_FORMULA);
+		rule1.setJustificatoin("C");
+		rule1.setConsequence("~D");
+
+		DefaultRule rule2 = new DefaultRule();
+		rule2.setPrerequisite(a.e.EMPTY_FORMULA);
+		rule2.setJustificatoin("D");
+		rule2.setConsequence("~E");
+
+		DefaultRule rule3 = new DefaultRule();
+		rule3.setPrerequisite(a.e.EMPTY_FORMULA);
+		rule3.setJustificatoin("E");
+		rule3.setConsequence("~F");
+
+		RuleSet myRules = new RuleSet();
+		myRules.addRule(rule1);
+		myRules.addRule(rule2);
+		myRules.addRule(rule3);
+
+		DefaultReasoner loader = new DefaultReasoner(myWorld, myRules);
+		HashSet<String> extensions = loader.getPossibleScenarios();
+
+		a.e.println("Given the world: \n\t" + myWorld.toString()
+				+ "\n And the rules \n\t" + myRules.toString());
+
+		a.e.println("Possible Extensions");
+		for (String c : extensions) {
+			a.e.println("\t Ext: Th(W U (" + c + "))");
+			// Added closure operator
+			a.e.incIndent();
+			WFF world_and_ext = new WFF("(( " + myWorld.getWorld() + " ) & ("
+					+ c + "))");
+			a.e.println("= " + world_and_ext.getClosure());
+			a.e.decIndent();
+
+		}
+	}
+
+	/**
+	 * Output Applyig rules to the world Effect:eeee & eeee ==== eeee Given the
+	 * world:
+	 * <p/>
+	 * And the rules [([]):(C) ==> (~D)] , [([]):(D) ==> (~C)] Possible
+	 * Extensions Ext: Th(W U (~D)) Ext: Th(W U (~C))
+	 */
+	public static void example8() {
+		WorldSet myWorld = new WorldSet(); // Empty World
+		myWorld.addFormula(a.e.EMPTY_FORMULA);
+
+		DefaultRule rule1 = new DefaultRule();
+		rule1.setPrerequisite(a.e.EMPTY_FORMULA);
+		rule1.setJustificatoin("C");
+		rule1.setConsequence("~D");
+
+		DefaultRule rule2 = new DefaultRule();
+		rule2.setPrerequisite(a.e.EMPTY_FORMULA);
+		rule2.setJustificatoin("D");
+		rule2.setConsequence("~C");
+
+		RuleSet myRules = new RuleSet();
+		myRules.addRule(rule1);
+		myRules.addRule(rule2);
+
+		DefaultReasoner loader = new DefaultReasoner(myWorld, myRules);
+		HashSet<String> extensions = loader.getPossibleScenarios();
+
+		a.e.println("Given the world: \n\t" + myWorld.toString()
+				+ "\n And the rules \n\t" + myRules.toString());
+
+		a.e.println("Possible Extensions");
+		for (String c : extensions) {
+			a.e.println("\t Ext: Th(W U (" + c + "))");
+			// Added closure operator
+			a.e.incIndent();
+			WFF world_and_ext = new WFF("(( " + myWorld.getWorld() + " ) & ("
+					+ c + "))");
+			a.e.println("= " + world_and_ext.getClosure());
+			a.e.decIndent();
+
+		}
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 584:617, 627:660.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.